shared/src/main/scala/amf/core/client/scala/AMFGraphConfiguration.scala

```diff
@@ -186,6 +190,9 @@ sealed abstract class BaseAMFConfigurationSetter(private[amf] val resolvers: AMF
   protected def _withPlugin[T](amfPlugin: AMFPlugin[_]): T =
     copy(registry = registry.withPlugin(amfPlugin)).asInstanceOf[T]
```

Comment:
I think default "withPlugin" method should registry the plugin as both possibilities, root and references.
Specific methods for only root and only reference can be added and exposed as particular alternatives.

Reply:
Agreed, this does imply that the majority of the parse plugins will be duplicated in both lists but gives a clear sense of how root and reference files will be handled. Adjusted here, https://github.com/aml-org/amf-aml/pull/442, and https://github.com/aml-org/amf/pull/1241.